### PR TITLE
Family album page 

### PR DESCRIPTION
**Pull request by SophieSjoberg**
### Description 
- User can share photos on a Family album page 
- Add gallery functionality with lightbox-bootstrap-rails gem 
- Adds association between user and album through family

## Screenshot 
screen shot 2018-02-03 at 18 56 21

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class User < ApplicationRecord
+  belongs_to :family, optional: true
```
Comment (NegativeKarma) [inline app/models/user.rb:2]: this ```optional: true``` means that it can belong to something but it can even not belong to that something?

**Review by NegativeKarma — APPROVED** [review]
LGTM

File: app/models/user.rb (hunk 1 above)
Comment (tochman) [reply app/models/user.rb:2]: Good job

File: Gemfile [hunk 2]
```diff
@@ -18,6 +18,8 @@ gem 'bootstrap', '~> 4.0'
 gem 'webpacker', git: 'https://github.com/rails/webpacker.git'
 gem 'react-rails'
 gem 'aws-sdk', '~> 2'
+gem 'lightbox-bootstrap-rails', '5.1.0.1'
+gem 'coffee-rails'
```
Comment (tochman) [inline Gemfile:22]: Why do you add coffee?

File: app/assets/javascripts/lightbox_bootstraped.coffee [hunk 3]
```diff
@@ -0,0 +1,4 @@
+$(document).delegate '*[data-toggle="lightbox"]', 'click', (event) ->
+  event.preventDefault()
+  $(this).ekkoLightbox()
+  return
```
Comment (tochman) [inline app/assets/javascripts/lightbox_bootstraped.coffee]: You can translate this to jQuery

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -8,8 +8,14 @@
 
     <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track': 'reload' %>
     <%= javascript_include_tag 'application', 'data-turbolinks-track': 'reload' %>
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/ekko-lightbox/5.3.0/ekko-lightbox.css " rel="stylesheet" type="text/css">
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/ekko-lightbox/5.3.0/ekko-lightbox.js  " rel="stylesheet" type="text/css">
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/ekko-lightbox/5.3.0/ekko-lightbox.js.map" rel="stylesheet" type="text/css">
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/ekko-lightbox/5.3.0/ekko-lightbox.min.js" rel="stylesheet" type="text/css">
+    <link href="https://cdnjs.cloudflare.com/ajax/libs/ekko-lightbox/5.3.0/ekko-lightbox.min.js.map" rel="stylesheet" type="text/css">
```
Comment (tochman) [inline app/views/layouts/application.html.erb]: Not sure about using cdn. Please look for a gem. 🤪

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -19,5 +25,11 @@
 
     <%= render 'partials/navbar' %>
     <%= yield %>
+    <script>
+    $(document).on('click', '[data-toggle="lightbox"]', function(event) {
+                event.preventDefault();
+                $(this).ekkoLightbox();
```
Comment (tochman) [inline app/views/layouts/application.html.erb]: Am I wrong or are you doing this twice?

File: db/migrate/20180203144744_add_family_albums_to_users.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+class AddFamilyAlbumsToUsers < ActiveRecord::Migration[5.1]
+  def self.up
+    change_table :users do |t|
+      t.belongs_to :family_album, foreign_key: true
```
Comment (tochman) [inline db/migrate/20180203144744_add_family_albums_to_users.rb]: Why do you need this association? I would look into has many through associations user -> family -> family_album

File: app/views/layouts/application.html.erb (hunk 5 above)
Comment (SophieSjoberg) [reply app/views/layouts/application.html.erb]: you're wrong

Comment (tochman) [reply app/views/layouts/application.html.erb]: Please take a look at the coffee file added to this PR.

**Review by NegativeKarma — APPROVED** [review]
LGTM

**Pull request merged**